schema-engine/connectors/schema-connector/src/warnings.rs

```diff
@@ -1,6 +1,131 @@
 //! Warnings generator for Introspection
 
-use std::fmt;
+use std::{collections::BTreeMap, fmt};
+
+trait Groupable<K, V> {
+    /// Groups the items in the collection by a key, specified by the concrete implementation.
+    /// For any key, the value is a **non-empty** vector.
+    fn group_by(&self) -> BTreeMap<K, Vec<&V>>;
+}
+
+/// A group of warnings that can be grouped by a key, which depends on the concretely
+/// instantiated type T.
+#[derive(Debug, PartialEq)]
+pub struct GroupBy<'a, T>(&'a Vec<T>);
+
+impl Groupable<String, ModelAndField> for GroupBy<'_, ModelAndField> {
+    fn group_by(&self) -> BTreeMap<String, Vec<&ModelAndField>> {
+        let mut result: BTreeMap<String, Vec<&ModelAndField>> = BTreeMap::new();
+
+        for item in self.0 {
+            result.entry(item.model.clone()).or_default().push(item);
+        }
+
+        result
+    }
+}
```

Comment:
I meant we could remove the whole trait, and replace the lines that just call `group_by()` with something like `let grouped: BTreeMap<_, _> = self.iter().map(|mf| (&mf.model, mf)).collect();`